Pers/vprokhorenko/homework 16

Traknar2142 opened the pull request:
Реализация дз №16

Traknar2142 committed:
Рыба дз 16

Traknar2142 committed:
Реализация дз №16

stvort commented on homework-16/src/main/java/ru/otus/homework/actuators/EmptyBookTableCheckHealthIndicator.java:
```diff
@@ -0,0 +1,39 @@
+        }
+    }
+
+    private boolean isBookRepoEmpty() {
+        List<Book> all = bookRepository.findAll();
+        return all.isEmpty();
+    }
```
Это будет несколько избыточно, загрузить потенциально несколько сотен тысяч книг, чтобы проверить, что ин есть

stvort commented on homework-16/src/main/java/ru/otus/homework/actuators/EmptyBookTableCheckHealthIndicator.java:
```diff
@@ -0,0 +1,39 @@
+
+@Component
+public class EmptyBookTableCheckHealthIndicator implements HealthIndicator {
+    private final BookRepository bookRepository;
```
У нас же есть сервисы вроде?

stvort commented on homework-16/src/main/resources/application.yml:
```diff
@@ -0,0 +1,40 @@
+  endpoints:
+    web:
+      exposure:
+        include: "*"
```
В задании нужно было подключить конкретные эндпойнты (+ свои если есть)
>Включить метрики, healthchecks и logfile.

Сейчас включены все

Traknar2142 committed:
- Сделал отображение только трех метрик: metrics,health,logfile
- убрал репозиторий из healthCheckera, добавил сервис
- переписал логику определения кол-ва книг в базе

stvort commented on homework-16/src/main/java/ru/otus/homework/service/BookServiceImpl.java:
```diff
@@ -71,6 +71,12 @@ public void deleteBook(Long id) {
         bookRepository.deleteById(id);
     }
 
+    @Transactional
```
Зачем тут транзакция?

Traknar2142 committed:
Небольшие правки
